tools/targets/STM32_gen_PeripheralPins.py

```diff
@@ -332,7 +334,12 @@ def store_qspi(pin, name, signal):
 
 # function to store USB pins
 def store_usb(pin, name, signal):
-    usb_list.append([pin, name, signal])
+    if "OTG" not in signal:
+        usb_list.append([pin, name, signal])
+    if signal.startswith("USB_OTG_FS"):
+        usb_otgfs_list.append([pin, name, signal])
+    if signal.startswith("USB_OTG_HS"):
+        usb_otghs_list.append([pin, name, signal])
```

Comment:
This could be incorrect as it changes the behaviour for all signals that contain 'OTG' that do not match the special case. Whilst there may be none of these today, this will fail to add the pin in future if the signal contains 'OTG'. Suggest an if..elif..elif...else construct could be better here.
If this is not the intent then there should be a big comment explaining why. Ideally there should be a comment to explain what this new code is trying to do in any case.